go/libraries/doltcore/sqle/schema.go

```diff
@@ -65,17 +69,31 @@ func SqlSchemaToDoltResultSchema(sqlSchema sql.Schema) (schema.Schema, error) {
 
 // SqlSchemaToDoltResultSchema returns a dolt Schema from the sql schema given, suitable for use in creating a table.
 // For result set schemas, see SqlSchemaToDoltResultSchema.
-func SqlSchemaToDoltSchema(sqlSchema sql.Schema) (schema.Schema, error) {
+func SqlSchemaToDoltSchema(ctx context.Context, root *doltdb.RootValue, tableName string, sqlSchema sql.Schema) (schema.Schema, error) {
 	var cols []schema.Column
+	var err error
+
+	nks, err := extractNomsKinds(sqlSchema)
+	if err != nil {
+		return nil, err
+	}
+	if len(sqlSchema) != len(nks) {
+		panic("slices must have equal len")
+	}
+	var existing []types.NomsKind
 
 	var tag uint64
 	for i, col := range sqlSchema {
 		commentTag := extractTag(col)
 		if commentTag != schema.InvalidTag {
 			tag = commentTag
 		} else {
-			tag = uint64(i)
+			tag, err = root.GetUniqueTagFromNomsKinds(ctx, tableName, col.Name, existing, nks[i])
```

Comment:
This logic makes me nervous. It's really easy for somebody to forget to update the existing kinds in the loop iteration, which would lead to different tag generation results, which would make repos affected by that bug incompatible with ones after we fixed it.

I think we need to make GetUniqueTag into a method that generates N tags as a batch. N can be 1 for the case of adding a single column to a table. But that at least makes it less likely someone will introduce this kind of error.

Reply:
This makes sense, I'll change it